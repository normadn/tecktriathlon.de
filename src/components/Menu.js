import React from "react";
import ScrollSpy from "react-scrollspy";

import colored from "../assets/img/logo/qhacksCrown-colored.svg";
import amsel from "../assets/img/logo/tvb-squared.png";

const Menu = (props) => {
  return (
    <div
      css={{
        display: "grid",
        gridTemplateColumns: "min-content auto",
        overflow: "visible",
        height: "84px"
      }}
    >
      <a
        href="/#"
        css={{
          transition: "0.5s",
          "@media(min-width:820px)": {
            display: props.standaloneVersion
              ? "inline-block"
              : props.imgCss
                ? "inline-block"
                : "none"
          }
        }}
      >
        <img
          css={{
            paddingTop: "20px",
            paddingBottom: "20px",
            height: "90px"
          }}
          src={amsel}
          alt="Amsel bird"
        />
      </a>
      <a
        id="mlh-trust-badge"
        href="https://amsel.tech"
        target="_blank"
        rel="external noopener"
        css={{
          display: "none",
          "@media(min-width:820px)": {
            display: props.standaloneVersion
              ? "none"
              : props.imgCss
                ? "none"
                : "inline-block"
          },
          transition: "0.5s"
        }}
      >
        
      </a>
      <div
        align="right"
        css={{
          paddingTop: "32px",
          paddingBottom: "32px",
          height: "84px",
          display: "grid",
          transition: "1s",
          marginLeft: "auto",
          "@media(max-width: 820px)": {
            display: "none"
          },
          " li": {
            listStyleType: "none"
          }
        }}
      >
        <ScrollSpy
          css={{
            display: "grid",
            gridColumnGap: "40px",
            gridTemplateColumns: props.menuItems
              .map(() => "min-content ")
              .reduce((a, b) => a + b)
          }}
          offset={-1}
          items={props.menuItems}
          currentClassName="is-current"
        >
          {props.menuItems.map((label) => (
            <li
              key={label}
              css={{
                ":hover": {
                  " a": {
                    color: "#00205b"
                  },
                  ">div": {
                    width: "100%",
                    height: "2px"
                  }
                }
              }}
            >
              <a
                css={{
                  lineHeight: "1.43",
                  textAlign: "center",
                  paddingLeft: "2px",
                  paddingRight: "2px",
                  textTransform: "uppercase"
                }}
                href={`/#${label}`}
              >
                {label}
              </a>
              <div
                css={{
                  width: "100%",
                  height: "2px",
                  transition: "0.5s",
                  marginLeft: "auto",
                  marginRight: "auto",
                  backgroundColor: "#00205b",
                  width: "0px"
                }}
              />
            </li>
          ))}
        </ScrollSpy>
      </div>
    </div>
  );
};

export default Menu;
