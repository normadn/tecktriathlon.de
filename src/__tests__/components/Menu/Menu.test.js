import React from "react";
import Menu from "../../../components/Menu";
import { shallow } from "enzyme";

describe("Menu", () => {
  it("renders correctly", () => {
    expect(
      shallow(<Menu menuItems={["THIS", "IS", "A", "TEST"]} />)
    ).toMatchSnapshot();
  });
});
