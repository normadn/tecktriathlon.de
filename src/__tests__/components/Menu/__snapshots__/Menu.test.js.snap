// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<div
  css={
    Object {
      "display": "grid",
      "gridTemplateColumns": "min-content auto",
      "height": "84px",
      "overflow": "visible",
    }
  }
>
  <a
    css={
      Object {
        "@media(min-width:820px)": Object {
          "display": "none",
        },
        "transition": "0.5s",
      }
    }
    href="/#"
  >
    <img
      alt="QHacks Crown"
      css={
        Object {
          "paddingBottom": "20px",
          "paddingTop": "20px",
        }
      }
      src="test-file-stub"
    />
  </a>
  <a
    css={
      Object {
        "@media(min-width:820px)": Object {
          "display": "inline-block",
        },
        "display": "none",
        "transition": "0.5s",
      }
    }
    href="https://mlh.io/seasons/na-2019/events?utm_source=na-hackathon&utm_medium=TrustBadge&utm_campaign=2019-season&utm_content=white"
    id="mlh-trust-badge"
    rel="external noopener"
    target="_blank"
  >
    <img
      alt="Major League Hacking 2019 Hackathon Season"
      css={
        Object {
          "height": "150px",
        }
      }
      src="https://s3.amazonaws.com/logged-assets/trust-badge/2019/mlh-trust-badge-2019-white.svg"
    />
  </a>
  <div
    align="right"
    css={
      Object {
        " li": Object {
          "listStyleType": "none",
        },
        "@media(max-width: 820px)": Object {
          "display": "none",
        },
        "display": "grid",
        "height": "84px",
        "marginLeft": "auto",
        "paddingBottom": "32px",
        "paddingTop": "32px",
        "transition": "1s",
      }
    }
  >
    <Scrollspy
      componentTag="ul"
      css={
        Object {
          "display": "grid",
          "gridColumnGap": "40px",
          "gridTemplateColumns": "min-content min-content min-content min-content ",
        }
      }
      currentClassName="is-current"
      items={
        Array [
          "THIS",
          "IS",
          "A",
          "TEST",
        ]
      }
      offset={-1}
      onUpdate={[Function]}
      style={Object {}}
    >
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="THIS"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="/#THIS"
        >
          THIS
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="IS"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="/#IS"
        >
          IS
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="A"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="/#A"
        >
          A
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="TEST"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="/#TEST"
        >
          TEST
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
    </Scrollspy>
  </div>
</div>
`;
